upgrade/v102to110/upgrade.go

```diff
@@ -0,0 +1,79 @@
+package v102to110
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+
+	lhclientset "github.com/longhorn/longhorn-manager/k8s/pkg/client/clientset/versioned"
+	"github.com/longhorn/longhorn-manager/types"
+)
+
+// This upgrade is needed because we add one more field `controller: true`
+// to the ownerReferences of instance manager pods so that `kubectl drain`
+// can work without --force flag.
+// Therefore, we need to updade the field for all existing instance manager pod
+// Link to the original issue: https://github.com/longhorn/longhorn/issues/1286
+
+const (
+	upgradeLogPrefix = "upgrade from v1.0.2 to v1.1.0: "
+)
+
+func UpgradeInstanceManagerPods(namespace string, lhClient *lhclientset.Clientset, kubeClient *clientset.Clientset) (err error) {
+	defer func() {
+		err = errors.Wrapf(err, upgradeLogPrefix+"upgrade instance manager pods failed")
+	}()
+
+	imPodsList, err := kubeClient.CoreV1().Pods(namespace).List(metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", types.GetLonghornLabelComponentKey(), types.LonghornLabelInstanceManager),
+	})
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return nil
+		}
+		return errors.Wrapf(err, upgradeLogPrefix+"failed to list all existing instance manager pods during the instance managers pods upgrade")
+	}
+
+	for _, pod := range imPodsList.Items {
+		if err := upgradeInstanceMangerPodOwnerRef(&pod, kubeClient, namespace); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func upgradeInstanceMangerPodOwnerRef(pod *v1.Pod, kubeClient *clientset.Clientset, namespace string) (err error) {
+	metadata, err := meta.Accessor(pod)
+	if err != nil {
+		return err
+	}
+
+	podOnwerRefs := metadata.GetOwnerReferences()
+	isController := true
+	needToUpdate := false
+	for ind, ownerRef := range podOnwerRefs {
```

Comment:
Need to check if it's owned by Longhorn, in case some other components are using the field as well.

Reply:
Thank you! Fixed